### PR TITLE
Fix crash in drawing when right channel has more clips than left

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5102

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-channel/Channel.h [hunk 1]
```diff
@@ -261,24 +261,19 @@ class CHANNEL_API Channel
    private:
       ChannelType *mpChannel{};
```
Comment (saintmatthieu) [inline libraries/lib-channel/Channel.h:262]: Couldn't these be const actually ?

**Review by saintmatthieu — COMMENTED** [review]
I understand this commit as a minimal, "preserve-from-crashing" commit that circumvents the fact that stereo tracks still can be made of inconsistent left/right clip counts, by always returning the count of the leader.
It seems to fix the situation where the right channel has most clips, but crash persists if left has most clips. I suppose this is because drawing thinks the right channel then has more clips than it actually has but still tries to draw all.
So might a better workaround be reporting the least clip count across channels ?
@Paul-Licameli you mentioned recently that disallowing joining tracks with inconsistent clip boundaries was still for relatively far away within the current refactoring effort ? Having this earlier than planned, even in a quick-and-dirty way, might still be less dirty than the situation we're trying to deal with now.

**Review by vsverchinsky — COMMENTED** [review]
Fix doesn't seem to be complete:
изображение

STR:
1) Create mono track and single clip to it
2) Duplicate track
3) Make a copy clip in the second track
4) Call "Make Stereo Track" on the first track

**Comment by Paul-Licameli** [comment]
This crash prevention isn't worth it now.

Let it remain a known issue until the right time for it in this plan #4681

**Pull request closed without merging**